xarray/core/indexes.py

```diff
@@ -1495,11 +1495,11 @@ def filter_indexes_from_coords(
     of coordinate names.
 
     """
-    filtered_indexes: dict[Any, Index] = dict(**indexes)
+    filtered_indexes: dict[Any, Index] = dict(indexes)
 
     index_coord_names: dict[Hashable, set[Hashable]] = defaultdict(set)
     for name, idx in indexes.items():
-        index_coord_names[id(idx)].add(name)
```

Comment:
This seems sketchy to me, but I'm also not familiar with this code. Can we revert please?

Reply:
Done.